idl/chimera.pro

```diff
@@ -0,0 +1,535 @@
+							props[9,ident+1]=string(strcompress(Xsb,/remove_all),format='(I10.0)')
+							props[10,ident+1]=string(strcompress(Ysb,/remove_all),format='(I10.0)')
+							props[11,ident+1]=string(strcompress(width,/remove_all),format='(I10.0)')
+							armm=string(strcompress((trummar/(1e+12)),/remove_all),format='(e11.1)')
+							props[12,ident+1]=strmid(armm,0,8)+strmid(armm,9,2)
+							props[13,ident+1]=string(strcompress((arcar*100/(!PI*rs^(2))),/remove_all),format='(F10.1)')
+							props[14,ident+1]=string(strcompress(mB,/remove_all),format='(F10.1)')
+							props[15,ident+1]=string(strcompress(mBpos,/remove_all),format='(F10.1)')
+							props[16,ident+1]=string(strcompress(mBneg,/remove_all),format='(F10.1)')
+							props[17,ident+1]=string(strcompress(max(magpol),/remove_all),format='(F10.1)')
+							props[18,ident+1]=string(strcompress(min(magpol),/remove_all),format='(F10.1)')
+							tbpos=string(strcompress(total(npix[where(magpol gt 0)]),/remove_all),format='(e11.1)')
+							props[19,ident+1]=strmid(tbpos,0,8)+strmid(tbpos,9,2)
+							tbneg=string('-'+strcompress(total(npix[where(magpol lt 0)]),/remove_all),format='(e10.1)')
+							props[20,ident+1]=strmid(tbneg,0,7)+strmid(tbneg,8,2)
+							phi=string(strcompress(mB*trummar*1e+16,/remove_all),format='(e11.1)')
+							props[21,ident+1]=strmid(phi,0,8)+strmid(phi,9,2)
+							phipos=string(strcompress(mBpos*trummar*1e+16,/remove_all),format='(e11.1)')
+							props[22,ident+1]=strmid(phipos,0,8)+strmid(phipos,9,2)
+							phineg=string('-'+strcompress(abs(mBneg*trummar*1e+16),/remove_all),format='(e10.1)')
+							props[23,ident+1]=strmid(phineg,0,7)+strmid(phineg,8,2)
+
+;=====sets up code for next possible coronal hole=====
+							ident=ident+1
+						endif else iarr[subscripts]=0
+					endif else iarr[subscripts]=0
+				endelse
+			endelse
+		endif
+	endif
+endfor
+
+;=====sets ident back to max value of iarr======
+ident=ident-1
+
+;=====looks for a previous segmentation array==========
+if keyword_set(track) then begin
+	fchim=findfile(track+'*ch_loc*.sav')
+	restore,fchim[n_elements(fchim)-1]
+	preseg=lonarr(s[1],s[2])
+
+	for i=0,n_elements(chim.ch)-1 do begin
+	preseg[(chim.x[total(chim.n[0:i]):(chim.n[i+1]+total(chim.n[0:i]))-1]-fix(min(xco)))/chim.index[1].cdelt1,(chim.y[total(chim.n[0:i]):(chim.n[i+1]+total(chim.n[0:i]))-1]-fix(min(yco)))/chim.index[1].cdelt2]=chim.ch[i]
+	endfor
+
+	mxseg=chim.mxseg
+	dayb=anytim(chim.index[1].date,/utime)
+endif else begin
+	preseg=lonarr(s[1],s[2])
+	dayb=anytim(ind[1].date,/utime)
+	mxseg=0
+endelse
+
+;======finds time difference for tracking======
+daya=anytim(ind[1].date,/utime)
+secs=daya-dayb
+diff=diff_rot(secs/86400.0,lat)
+prevrot=lonarr(s[1],s[2])
+cenarr=intarr(s[1],s[2])
+
+;=====only track if previous segmentation given=======
+if max(preseg) gt 0 then begin
+for i=min(preseg[where(preseg gt 0)]),max(preseg) do begin
+
+;=====calculate centroids of old segmentation=======
+	cenarr[*]=0
+	wh=where(preseg eq (i))
```

Comment:
This where could be set outside the loop using histogram. It will make the whole thing running faster